### PR TITLE
[4.10] CLOUDSTACK-7985: assignVM in Advanced zone with Security Groups

### PR DESCRIPTION
**Pull request by ustcweizhou**
This commit contains the following changes:
(1) implementation of assignVM in Advanced zone with Security Groups
(2) keep the default nic on shared network when assignVM
(3) allow migrate vm from/to project;
(4) UI change for selecting account/project/network

### CODE REVIEW

**Comment by miguelaferreira** [comment]
Apache Jenkins build reports 5 failing tests (https://builds.apache.org/job/cloudstack-pull-analysis/575/testReport/)
 
screen shot 2015-09-16 at 20 24 56

**Comment by DaanHoogland** [comment]
@ustcweizhou @miguelaferreira @karuturi @remibergsma so no merges till this is fixed?

**Comment by wilderrodrigues** [comment]
@ustcweizhou ,

Which tests did you execute?

I would say: no merge until it's fixed, tested by @ustcweizhou and he let us know how we should be able to test.

Cheers,
Wilder

**Comment by DaanHoogland** [comment]
@wilderrodrigues that is not fair, it is not @ustcweizhou 's bug that needs fixing!

**Comment by miguelaferreira** [comment]
@DaanHoogland @wilderrodrigues @ustcweizhou  I agree with Daan, the problem is most likely elsewhere. But I also agree with Wilder, first we need to figure out where the problem is, fix it, rebase all PRs to branch off after the fix and then continue the normal process. We can't be merging things hoping all will be well.

**Comment by miguelaferreira** [comment]
Thanks for rebasing @ustcweizhou 

The pull-analysis is now running.
https://builds.apache.org/job/cloudstack-pull-analysis/583/
Lets keep an eye on it, hopefully the problem will be gone.

**Comment by ustcweizhou** [comment]
@DaanHoogland @miguelaferreira @wilderrodrigues 
I have rebase with the lastest master, and force push to github. 
let's see if it will be fixed.

**Comment by miguelaferreira** [comment]
The build job already passed the point it had failed before (fingers-crossed)

**Comment by wilderrodrigues** [comment]
Hi @ustcweizhou ,

I will :-1:  this PR until we get to know which integration tests you executed, against which environment and the steps so we can do the same.

Cheers,
Wilder

**Comment by ustcweizhou** [comment]
@wilderrodrigues 

test cases
1. advanced zone
	(1) a shared network for domain domain1.
		account1: in domain 1, one isolated network
		acocunt2: in domain 1, two isolated networks
		project1: in domain 1, no network
		account3: in domain 2, one isolated network.
		
	(2) assign vm1 (account1, isolated network) to account2. expectation: fail, new owner is account2, but no network attached to the vm after assignment.
		assign vm2 (accoung2, isolated network) to account1. expectation: succeed. new owner is account1, one isolated network attached to vm.
		assign vm3 (account1, shared network) to account2.  expectation: succeed. new owner is account2, shared network attached to vm.
		assign vm4 (account1, shared network) to account3.  expectation: succeed. new owner is account3, one isolated network attached to vm.
		assign vm5 (account1, shared network) to project1.  expectation: succeed. new owner is project1, one isolated network created and attached to vm

2. advanced zone with security groups
	(1) a shared network 1 for domain domain1.
		a shared network 2 for domain domain2.
		account1: in domain 1
		account2: in domain 1
		account3: in domain 2
	(2) assign vm1 ((account1, shared network) to account2.  expectation: succeed. new owner is account2, shared network 1 attached to vm, security group applied.
		assign vm1 ((account1, shared network) to account3.  expectation: succeed. new owner is account3, shared network 2 attached to vm, security group applied.

**Comment by ustcweizhou** [comment]
guys, anyone has update ?

**Comment by wilderrodrigues** [comment]
Hi @ustcweizhou 

Thanks for the details. Are there some integration tests that we could use to cover the cases you described?

I can runt the following tests to make sure it doesn't break existing features:

```
echo "Running tests with required_hardware=true"
nosetests --with-marvin --marvin-config=[MARVIN_CONFIG_HERE] -s -a tags=advanced,required_hardware=true \
component/test_vpc_redundant.py \
component/test_routers_iptables_default_policy.py \
component/test_vpc_router_nics.py

echo "Running tests with required_hardware=false"
nosetests --with-marvin --marvin-config=[MARVIN_CONFIG_HERE] -s -a tags=advanced,required_hardware=false \
smoke/test_routers.py \
smoke/test_network_acl.py \
smoke/test_privategw_acl.py \
smoke/test_reset_vm_on_reboot.py \
smoke/test_vm_life_cycle.py \
smoke/test_vpc_vpn.py \
smoke/test_service_offerings.py \
component/test_vpc_offerings.py \
component/test_vpc_routers.py
```

But I'm not sure if those tests will cover all of your changes.

Cheers,
Wilder

**Comment by wilderrodrigues** [comment]
Hi @ustcweizhou @DaanHoogland @remibergsma @karuturi @miguelaferreira 

I executed the tests on my side. Despite some exceptions with network cleanup and the VPC tests that will pass only after PR #908 gets merged, everything went fine.

This PR L1/2GTM (Look 1/2 Good to Me) taking into account that it doesn't break existing features. I need more time to tests the security groups bit - or @ustcweizhou can test it  manually, or perhaps someone else can do help on that.

Results:

```
Create a redundant vpc with two networks with two vms in each network ... === TestName: test_01a_create_redundant_VPC | Status : FAILED ===
FAIL
=== TestName: test_01a_create_redundant_VPC | Status : EXCEPTION ===
ERROR
Test iptables default INPUT/FORWARD policy on RouterVM ... === TestName: test_02_routervm_iptables_policies | Status : SUCCESS ===
ok
Test iptables default INPUT/FORWARD policies on VPC router ... === TestName: test_01_single_VPC_iptables_policies | Status : SUCCESS ===
ok
=== TestName: test_01_single_VPC_iptables_policies | Status : EXCEPTION ===
ERROR
Create a vpc with two networks with two vms in each network ... === TestName: test_01_VPC_nics_after_destroy | Status : SUCCESS ===
ok
=== TestName: test_01_VPC_nics_after_destroy | Status : EXCEPTION ===
ERROR

Test router internal advanced zone ... === TestName: test_02_router_internal_adv | Status : SUCCESS ===
ok
Test restart network ... === TestName: test_03_restart_network_cleanup | Status : SUCCESS ===
ok
Test router basic setup ... === TestName: test_05_router_basic | Status : SUCCESS ===
ok
Test router advanced setup ... === TestName: test_06_router_advanced | Status : SUCCESS ===
ok
Test stop router ... === TestName: test_07_stop_router | Status : SUCCESS ===
ok
Test start router ... === TestName: test_08_start_router | Status : SUCCESS ===
ok
Test reboot router ... === TestName: test_09_reboot_router | Status : SUCCESS ===
ok
test_privategw_acl (integration.smoke.test_privategw_acl.TestPrivateGwACL) ... === TestName: test_privategw_acl | Status : SUCCESS ===
ok
Test reset virtual machine on reboot ... === TestName: test_01_reset_vm_on_reboot | Status : SUCCESS ===
ok
Test advanced zone virtual router ... === TestName: test_advZoneVirtualRouter | Status : SUCCESS ===
ok
Test Deploy Virtual Machine ... === TestName: test_deploy_vm | Status : SUCCESS ===
ok
Test Multiple Deploy Virtual Machine ... === TestName: test_deploy_vm_multiple | Status : SUCCESS ===
ok
Test Stop Virtual Machine ... === TestName: test_01_stop_vm | Status : SUCCESS ===
ok
Test Start Virtual Machine ... === TestName: test_02_start_vm | Status : SUCCESS ===
ok
Test Reboot Virtual Machine ... === TestName: test_03_reboot_vm | Status : SUCCESS ===
ok
Test destroy Virtual Machine ... === TestName: test_06_destroy_vm | Status : SUCCESS ===
ok
Test recover Virtual Machine ... === TestName: test_07_restore_vm | Status : SUCCESS ===
ok
Test migrate VM ... SKIP: At least two hosts should be present in the zone for migration
Test destroy(expunge) Virtual Machine ... === TestName: test_09_expunge_vm | Status : SUCCESS ===
ok
Test VPN in VPC ... === TestName: test_vpc_remote_access_vpn | Status : SUCCESS ===
ok
Test VPN in VPC ... === TestName: test_vpc_site2site_vpn | Status : SUCCESS ===
ok
Test to create service offering ... === TestName: test_01_create_service_offering | Status : SUCCESS ===
ok
Test to update existing service offering ... === TestName: test_02_edit_service_offering | Status : SUCCESS ===
ok
Test to delete service offering ... === TestName: test_03_delete_service_offering | Status : SUCCESS ===
ok
Test create VPC offering ... === TestName: test_01_create_vpc_offering | Status : SUCCESS ===
ok
Test VPC offering without load balancing service ... === TestName: test_03_vpc_off_without_lb | Status : EXCEPTION ===
ERROR
Test VPC offering without static NAT service ... === TestName: test_04_vpc_off_without_static_nat | Status : EXCEPTION ===
ERROR
Test VPC offering without port forwarding service ... === TestName: test_05_vpc_off_without_pf | Status : EXCEPTION ===
ERROR
Test VPC offering with invalid services ... === TestName: test_06_vpc_off_invalid_services | Status : SUCCESS ===
ok
Test update VPC offering ... === TestName: test_07_update_vpc_off | Status : SUCCESS ===
ok
Test list VPC offering ... === TestName: test_08_list_vpc_off | Status : SUCCESS ===
ok
test_09_create_redundant_vpc_offering (integration.component.test_vpc_offerings.TestVPCOffering) ... === TestName: test_09_create_redundant_vpc_offering | Status : SUCCESS ===
ok
Test start/stop of router after addition of one guest network ... === TestName: test_01_start_stop_router_after_addition_of_one_guest_network | Status : SUCCESS ===
ok
Test reboot of router after addition of one guest network ... === TestName: test_02_reboot_router_after_addition_of_one_guest_network | Status : SUCCESS ===
ok
Test to change service offering of router after addition of one guest network ... === TestName: test_04_chg_srv_off_router_after_addition_of_one_guest_network | Status : SUCCESS ===
ok
Test destroy of router after addition of one guest network ... === TestName: test_05_destroy_router_after_addition_of_one_guest_network | Status : SUCCESS ===
ok
Test to stop and start router after creation of VPC ... === TestName: test_01_stop_start_router_after_creating_vpc | Status : SUCCESS ===
ok
Test to reboot the router after creating a VPC ... === TestName: test_02_reboot_router_after_creating_vpc | Status : SUCCESS ===
ok
Tests to change service offering of the Router after ... === TestName: test_04_change_service_offerring_vpc | Status : SUCCESS ===
ok
Test to destroy the router after creating a VPC ... === TestName: test_05_destroy_router_after_creating_vpc | Status : SUCCESS ===
ok
```

**Comment by ustcweizhou** [comment]
@wilderrodrigues thanks for you testing. I believe this PR will not impact the current network/vr/storage.
It will only change the owner of a vm, and add the vm to a network /security group if applicable.

**Comment by miguelaferreira** [comment]
@ustcweizhou may I ask you if you believe that the change in this PR will not impact the areas you mention because you've tested it (been manually?). Or because you assume that this change is isolated from hose areas of the system?

We have had production issues in the past due to "isolated" changes to plugins for hypervisors we don't even use. Therefore, the urge for testing!

**Comment by ustcweizhou** [comment]
@miguelaferreira I have tested the test cases listed above. I think it is isolated from other areas.

I agree more and careful testing are better. Can you show more details in your issues so maybe we can get some lessons from them?

**Comment by miguelaferreira** [comment]
@ustcweizhou It is not clear to me if you tested the same things @wilderrodrigues did, or if you tested "the security groups bit". Would you be so kind as to make that explicit.

Regarding the remark I made about seemingly isolated code: the PR that fixed the issue we had in production is #761. If you are interested, pleas have a look at the problem description and solution (arguably no the best!) in the PR comments.

**Comment by ustcweizhou** [comment]
@miguelaferreira 
1. no, I did not test the same as @wilderrodrigues .
2.  I've tested advanced zone (without security groups), it works fine. For Advanced zone with security group, I used this code in 4.2.1 for long time, then I ported to mainstream.sorry I did not test it in advanced zone with SG. However, compared to the implementation for Basic zone, I think this code is good.
3. Different from PR #761 which might be called by other functions, this PR is just an implementation for AssignVM in advanced zone with SG. The worst problem I can see is that this functionality does not work, but it will never impact other functionalities.

**Comment by miguelaferreira** [comment]
Thanks for clarifying @ustcweizhou 
I guess that we then need to wait for when @wilderrodrigues will have time to test the SGs.

**Comment by ustcweizhou** [comment]
no problem for me.
it is always good to get feedback from community. @miguelaferreira 
I have some features/improvements to be ported from old CS to mainstream.
Looking forward to getting more feedback from you guys in the future.

**Comment by DaanHoogland** [comment]
@miguelaferreira @remibergsma are you satisfied with this PR, now?

**Comment by miguelaferreira** [comment]
@DaanHoogland I think the status is still the same as of 7 days ago. Or am I missing something?

**Comment by DaanHoogland** [comment]
So that means the SG tests are missing still. Ok, I will add it to my testing backlog.

**Comment by DaanHoogland** [comment]
did the regular regression suite in my sbp provided bubble and all passed.

network:
```
Create a redundant VPC with two networks with two VMs in each network ... === TestName: test_01_create_redundant_VPC_2tiers_4VMs_4IPs_4PF_ACL | Status : SUCCESS ===
ok
Create a redundant VPC with two networks with two VMs in each network and check default routes ... === TestName: test_02_redundant_VPC_default_routes | Status : SUCCESS ===
ok
Test iptables default INPUT/FORWARD policy on RouterVM ... === TestName: test_02_routervm_iptables_policies | Status : SUCCESS ===
ok
Test iptables default INPUT/FORWARD policies on VPC router ... === TestName: test_01_single_VPC_iptables_policies | Status : SUCCESS ===
ok
Stop existing router, add a PF rule and check we can access the VM ... === TestName: test_isolate_network_FW_PF_default_routes | Status : SUCCESS ===
ok
Test redundant router internals ... === TestName: test_RVR_Network_FW_PF_SSH_default_routes | Status : SUCCESS ===
ok
Create a VPC with two networks with one VM in each network and test nics after destroy ... === TestName: test_01_VPC_nics_after_destroy | Status : SUCCESS ===
ok
Create a VPC with two networks with one VM in each network and test default routes ... === TestName: test_02_VPC_default_routes | Status : SUCCESS ===
ok
Check the password file in the Router VM ... === TestName: test_isolate_network_password_server | Status : SUCCESS ===
ok
Check that the /etc/dhcphosts.txt doesn't contain duplicate IPs ... === TestName: test_router_dhcphosts | Status : SUCCESS ===
ok
Test to create Load balancing rule with source NAT ... === TestName: test_01_create_lb_rule_src_nat | Status : SUCCESS ===
ok
Test to create Load balancing rule with non source NAT ... === TestName: test_02_create_lb_rule_non_nat | Status : SUCCESS ===
ok
Test for assign & removing load balancing rule ... === TestName: test_assign_and_removal_lb | Status : SUCCESS ===
ok
Test to verify access to loadbalancer haproxy admin stats page ... === TestName: test02_internallb_haproxy_stats_on_all_interfaces | Status : SUCCESS ===
ok
Test create, assign, remove of an Internal LB with roundrobin http traffic to 3 vm's ... === TestName: test_01_internallb_roundrobin_1VPC_3VM_HTTP_port80 | Status : SUCCESS ===
ok
Test SSVM Internals ... === TestName: test_03_ssvm_internals | Status : SUCCESS ===
ok
Test CPVM Internals ... === TestName: test_04_cpvm_internals | Status : SUCCESS ===
ok
Test stop SSVM ... === TestName: test_05_stop_ssvm | Status : SUCCESS ===
ok
Test stop CPVM ... === TestName: test_06_stop_cpvm | Status : SUCCESS ===
ok
Test reboot SSVM ... === TestName: test_07_reboot_ssvm | Status : SUCCESS ===
ok
Test reboot CPVM ... === TestName: test_08_reboot_cpvm | Status : SUCCESS ===
ok
Test destroy SSVM ... === TestName: test_09_destroy_ssvm | Status : SUCCESS ===
ok
Test destroy CPVM ... === TestName: test_10_destroy_cpvm | Status : SUCCESS ===
ok
Test for port forwarding on source NAT ... === TestName: test_01_port_fwd_on_src_nat | Status : SUCCESS ===
ok
Test for port forwarding on non source NAT ... === TestName: test_02_port_fwd_on_non_src_nat | Status : SUCCESS ===
ok
Test for reboot router ... === TestName: test_reboot_router | Status : SUCCESS ===
ok
Test for Router rules for network rules on acquired public IP ... === TestName: test_network_rules_acquired_public_ip_1_static_nat_rule | Status : SUCCESS ===
ok
Test for Router rules for network rules on acquired public IP ... === TestName: test_network_rules_acquired_public_ip_2_nat_rule | Status : SUCCESS ===
ok
Test for Router rules for network rules on acquired public IP ... === TestName: test_network_rules_acquired_public_ip_3_Load_Balancer_Rule | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 29 tests in 13041.270s

OK
```
and vpc:
```
Test router internal advanced zone ... === TestName: test_02_router_internal_adv | Status : SUCCESS ===
ok
Test restart network ... === TestName: test_03_restart_network_cleanup | Status : SUCCESS ===
ok
Test router basic setup ... === TestName: test_05_router_basic | Status : SUCCESS ===
ok
Test router advanced setup ... === TestName: test_06_router_advanced | Status : SUCCESS ===
ok
Test stop router ... === TestName: test_07_stop_router | Status : SUCCESS ===
ok
Test start router ... === TestName: test_08_start_router | Status : SUCCESS ===
ok
Test reboot router ... === TestName: test_09_reboot_router | Status : SUCCESS ===
ok
test_privategw_acl (integration.smoke.test_privategw_acl.TestPrivateGwACL) ... === TestName: test_privategw_acl | Status : SUCCESS ===
ok
Test reset virtual machine on reboot ... === TestName: test_01_reset_vm_on_reboot | Status : SUCCESS ===
ok
Test advanced zone virtual router ... === TestName: test_advZoneVirtualRouter | Status : SUCCESS ===
ok
Test Deploy Virtual Machine ... === TestName: test_deploy_vm | Status : SUCCESS ===
ok
Test Multiple Deploy Virtual Machine ... === TestName: test_deploy_vm_multiple | Status : SUCCESS ===
ok
Test Stop Virtual Machine ... === TestName: test_01_stop_vm | Status : SUCCESS ===
ok
Test Start Virtual Machine ... === TestName: test_02_start_vm | Status : SUCCESS ===
ok
Test Reboot Virtual Machine ... === TestName: test_03_reboot_vm | Status : SUCCESS ===
ok
Test destroy Virtual Machine ... === TestName: test_06_destroy_vm | Status : SUCCESS ===
ok
Test recover Virtual Machine ... === TestName: test_07_restore_vm | Status : SUCCESS ===
ok
Test migrate VM ... SKIP: At least two hosts should be present in the zone for migration
Test destroy(expunge) Virtual Machine ... === TestName: test_09_expunge_vm | Status : SUCCESS ===
ok
Test Remote Access VPN in VPC ... === TestName: test_vpc_remote_access_vpn | Status : SUCCESS ===
ok
Test VPN in VPC ... === TestName: test_vpc_site2site_vpn | Status : SUCCESS ===
ok
Test to create service offering ... === TestName: test_01_create_service_offering | Status : SUCCESS ===
ok
Test to update existing service offering ... === TestName: test_02_edit_service_offering | Status : SUCCESS ===
ok
Test to delete service offering ... === TestName: test_03_delete_service_offering | Status : SUCCESS ===
ok
Test for delete account ... === TestName: test_delete_account | Status : SUCCESS ===
ok
Test for Associate/Disassociate public IP address for admin account ... === TestName: test_public_ip_admin_account | Status : SUCCESS ===
ok
Test for Associate/Disassociate public IP address for user account ... === TestName: test_public_ip_user_account | Status : SUCCESS ===
ok
Test for release public IP address ... === TestName: test_releaseIP | Status : SUCCESS ===
ok
Test create VPC offering ... === TestName: test_01_create_vpc_offering | Status : SUCCESS ===
ok
Test VPC offering without load balancing service ... === TestName: test_03_vpc_off_without_lb | Status : SUCCESS ===
ok
Test VPC offering without static NAT service ... === TestName: test_04_vpc_off_without_static_nat | Status : SUCCESS ===
ok
Test VPC offering without port forwarding service ... === TestName: test_05_vpc_off_without_pf | Status : SUCCESS ===
ok
Test VPC offering with invalid services ... === TestName: test_06_vpc_off_invalid_services | Status : SUCCESS ===
ok
Test update VPC offering ... === TestName: test_07_update_vpc_off | Status : SUCCESS ===
ok
Test list VPC offering ... === TestName: test_08_list_vpc_off | Status : SUCCESS ===
ok
test_09_create_redundant_vpc_offering (integration.component.test_vpc_offerings.TestVPCOffering) ... === TestName: test_09_create_redundant_vpc_offering | Status : SUCCESS ===
ok
Test start/stop of router after addition of one guest network ... === TestName: test_01_start_stop_router_after_addition_of_one_guest_network | Status : SUCCESS ===
ok
Test reboot of router after addition of one guest network ... === TestName: test_02_reboot_router_after_addition_of_one_guest_network | Status : SUCCESS ===
ok
Test to change service offering of router after addition of one guest network ... === TestName: test_04_chg_srv_off_router_after_addition_of_one_guest_network | Status : SUCCESS ===
ok
Test destroy of router after addition of one guest network ... === TestName: test_05_destroy_router_after_addition_of_one_guest_network | Status : SUCCESS ===
ok
Test to stop and start router after creation of VPC ... === TestName: test_01_stop_start_router_after_creating_vpc | Status : SUCCESS ===
ok
Test to reboot the router after creating a VPC ... === TestName: test_02_reboot_router_after_creating_vpc | Status : SUCCESS ===
ok
Tests to change service offering of the Router after ... === TestName: test_04_change_service_offerring_vpc | Status : SUCCESS ===
ok
Test to destroy the router after creating a VPC ... === TestName: test_05_destroy_router_after_creating_vpc | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 44 tests in 8842.630s

OK (SKIP=1)
```

the skip is because for migration at least 2 hosts are needed

LGTM

**Comment by bhaisaab** [comment]
@ustcweizhou Can you please meld the commits into a single commit, and incorporate any remaining changes asked in the review?

Ping, can we have a re-review

**Comment by ustcweizhou** [comment]
rebased, and fixed an issue in assignvm to project.

File: client/WEB-INF/classes/resources/messages.properties [hunk 1]
```diff
@@ -1823,6 +1823,7 @@ message.after.enable.swift=Swift configured. Note\: When you leave this page, yo
 message.alert.state.detected=Alert state detected
 message.allow.vpn.access=Please enter a username and password of the user that you want to allow VPN access.
 message.apply.snapshot.policy=You have successfully updated your current snapshot policy.
+message.assign.instance.another=Please specify the account type, domain, account name and network (optional) of the new account. <br> If the default nic of the vm is on a shared network, CloudStack will check if the network can be used by the new account if you do not specify one network. <br> If the default nic of the vm is on a isolated network, and the new account has more one isolated networks, you should specify one.
```
Comment (GabrielBrascher) [inline client/WEB-INF/classes/resources/messages.properties]: Hi @ustcweizhou, just pointing a small typo: "on **a** isolated network" => "on **an** isolated network". Thanks.

**Comment by rhtyd** [comment]
@ustcweizhou please rebase against latest master and push -f, update on status of your PR

**Comment by bvbharatk** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 117
 Hypervisor xenserver
 NetworkType Advanced
 Passed=73
 Failed=0
 Skipped=3

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0


**Failed tests:**

**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_static_role_account_acls
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vpc_vpn.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_routers.py
test_reset_vm_on_reboot.py
test_snapshots.py
test_deploy_vms_with_varied_deploymentplanners.py
test_login.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_disk_offerings.py

**Comment by ustcweizhou** [comment]
rebased with latest master

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by karuturi** [comment]
merging

**Pull request merged**